API document.md

```diff
@@ -0,0 +1,832 @@
+# API document
+
+## Login related
+
+###  The project will use JWT tokens to achieve session tracking and login authentication，please store the JWT token in the *local storage* when login for the first time. In the subsequent requests, please add the token into the request header in the format *token：“JWT”*.
+
+### the token will record the userId，and the signature algorithm is HS256。（HEADER and PAYLOAD can be directly decoded by Base64）==users can access other functions except login only after the token has been successfully verified.==
+
+### If token resolution fails, which means the token is expired or tampered with, the server will return the following message
+
+```json
+{
+    "code":0,
+    "msg":"NOT_LOGIN",
+    "data":null
+}
+```
+
+### ==***<u>Please let the user login again by jumping to the login page when front-end receive the msg as above.</u>***==
+
+### 1. Login
+
+- request path: /login
+
+- request method: POST
+
+- request arguments type: Json
+
+  - name: must
+
+  - password: must
+
+  - ```json
+    //sample
+    
+    {
+        "name":"test",
+        "password":"123qwe"
+    }
+    ```
+
+- response:
+
+  - ```json
+    //sample
+    
+    {
+        "code":1,
+        "msg":"success",
+        "data":"eyJhbGciOiJIUzI1NiJ9.eyJuYW1lIjoibW95dWVyIiwiaWQiOjEsImV4cCI6MTcyMzcxMjQ2NX0.2pW4GSyLR_HiB2GbU3wkQfTs4HgwuKtWGUeVqHBaEVs" 
+    }
+    ```
+
+
+
+### 2. Sign up
+
+- request path: /login
+
+- request method: PUT
+
+- request arguments type: json
+
+  - name: must
+
+  - password: must
```

Comment:
Is there any requirements on the password? i.e. min length, capital letters or numbers needed etc. If so, do you think this should be displayed on the sign up page?

Reply:
I think the password should be at least 6 characters. Other requirements are unnecessary. The dynamic detection of password format is needed to be realized on sign up page by front-end.

Reply:
@RuipuCui Can you pls look into this further since you are responsible for the sign up page?